gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPickerViewModel.kt

```diff
@@ -77,7 +77,47 @@ internal class AvatarPickerViewModel(
             AvatarPickerEvent.DownloadManagerDisabledDialogDismissed -> hideDownloadManagerAlert()
             is AvatarPickerEvent.AvatarDeleteSelected -> deleteAvatar(event.avatarId)
             AvatarPickerEvent.AvatarDeleteAlertDismissed -> hideNonSelectedAvatarAlert()
-            is AvatarPickerEvent.AvatarRatingSelected -> Unit
+            is AvatarPickerEvent.AvatarRatingSelected -> updateAvatar(event.avatarId, event.rating)
+        }
+    }
+
+    private fun updateAvatar(avatarId: String, rating: Avatar.Rating? = null, altText: String? = null) {
+        viewModelScope.launch {
+            val oldAvatar: Avatar? = _uiState.value.emailAvatars?.avatars?.find { it.imageId == avatarId }
+            val updateType = AvatarUpdateType.RATING
```

Comment:
We need to handle the `updateType` when adding the `altText` but we'll do it in its own PR. :)

Reply:
Yes, I had it here, but I noticed I was going out of the scope of this PR, so I removed it.